packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -390,19 +392,77 @@ class FlutterPlugin implements Plugin<Project> {
      * Configures the Flutter plugin dependencies.
      *
      * The plugins are added to pubspec.yaml. Then, upon running `flutter pub get`,
-     * the tool generates a `.flutter-plugins` file, which contains a 1:1 map to each plugin location.
+     * the tool generates a `.flutter-plugins-dependencies` file, which contains a map to each plugin location.
      * Finally, the project's `settings.gradle` loads each plugin's android directory as a subproject.
      */
-    private void configurePlugins() {
-        getPluginList().each this.&configurePluginProject
-        getPluginDependencies().each this.&configurePluginDependencies
+    private void configurePlugins(Project project) {
+        configureLegacyPluginEachProjects(project)
+        getPluginList(project).each this.&configurePluginProject
+        getPluginList(project).each this.&configurePluginDependencies
+    }
+
+    // TODO(54566, 48918): Can remove once the issues are resolved.
```

Comment:
nit: modify this and other todos to match the style guide format described in https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions

Reply:
@gmackall this was proposed to me. I had to remove the hashtags, to not annoy the linter.

See:
https://github.com/flutter/flutter/pull/137115#discussion_r1388573276 
https://github.com/flutter/flutter/pull/137115#discussion_r1425495828

Or do you mean the wording?

Reply:
Ah, sorry I missed that it was suggested in the comment history. I'm fine to leave it as is given that.